app/main.py

```diff
@@ -85,6 +93,30 @@ def create_app() -> FastAPI:
     return app
 
 
+def get_auth_deps() -> list[Depends]:
+    "function in order to fetch dependencies for authentication"
+    deps = os.environ.get("OTEAPI_AUTH_DEPS")
+    if deps:
+        mods = [
+            module.replace(" ", str()).rpartition(".") for module in deps.split("|")
+        ]
+        try:
+            imports = [
+                getattr(import_module(module), classname)
+                for (module, dot, classname) in mods
+            ]
+        except Exception as error:
+            raise error
+        logger.info(
+            "Imported the following dependencies for authentication: %s", imports
+        )
+        dependencies = [Depends(dependency()) for dependency in imports]
```

Comment:
I am not sure here what is the convention. For testing, I used a derivation of the [HTTPBearer](https://gitlab.cc-asp.fraunhofer.de/MarketPlace/platform/modules/accesscontrolpackage/-/blob/enh/fastapi/src/accesscontrol/frameworks/fastapi.py#L19) from the MarketPlace which overriding the `__call__`-method for the callback to the OpenID-Client. Do you think it is more convenient to perform the callback in the `__init__`-method or even the `__post__init__` so that the dependency does not have to called in `Depends`?